predicators/perception/sokoban_perceiver.py

```diff
@@ -0,0 +1,97 @@
+"""A sokoban-specific perceiver."""
+
+from typing import Dict, Tuple
+
+import numpy as np
+
+from predicators import utils
+from predicators.envs.sokoban import SokobanEnv
+from predicators.perception.base_perceiver import BasePerceiver
+from predicators.structs import EnvironmentTask, GroundAtom, Object, \
+    Observation, State, Task
+
+# Each observation is a tuple of four 2D boolean masks (numpy arrays).
+# The order is: free, goals, boxes, player.
+
+
+class SokobanPerceiver(BasePerceiver):
```

Comment:
Great work on this! I LOVE how this perception is now separated from the env, and am very glad we decided to implement `Sokoban`! 😸